src/error.rs

```diff
@@ -0,0 +1,43 @@
+use platform_impl;
+
```

Comment:
Can you implement `std::error::Error` for this so that conversion with `?` will work?

Reply:
That's the plan. I didn't do it for the initial PR since I was on a machine where that wasn't easy to do (platform-specific `Display` implementations are necessary), but those should definitely get in before this gets merged.